server/plugin/plugin.go

```diff
@@ -783,13 +784,64 @@ func (p *Plugin) isOrganizationLocked() bool {
 }
 
 func (p *Plugin) sendRefreshEvent(userID string) {
+	eventLogger := logger.New(p.API).With(logger.LogContext{
+		"userid": userID,
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+
+	context := &Context{
+		Ctx:    ctx,
+		UserID: userID,
+		Log:    eventLogger,
+	}
+
+	defer cancel()
+
+	info, apiErr := p.getGitHubUserInfo(context.UserID)
+	if apiErr != nil {
+		p.API.LogWarn("Failed to get github user info", "error", apiErr.Error())
+		return
+	}
+
+	context.Log = context.Log.With(logger.LogContext{
+		"github username": info.GitHubUsername,
+	})
+
+	userContext := &UserContext{
+		Context: *context,
+		GHInfo:  info,
+	}
+
+	sidebarContent := p.getSidebarData(userContext)
+
+	contentMap, err := convertContentToMap(sidebarContent)
```

Comment:
same here, do we need to check if `sidebarContent` is `nil` in case of any API failure for the APIs called inside `getSidebarData`?